lib/wibox/layout/align.lua

```diff
@@ -35,127 +35,336 @@ local base = require("wibox.widget.base")
-                y = floor( (height - h)/2 )
+        elseif max_size_side * 2 + size_second >= size_total then
+            -- Could place a bit of the second widget
+            -- If do not fully expand the smaller widget
+
+            if first_larger then
+                place_widget(
+                    -- At the beginning
+                    first,
+                    -- Takes the space it needs
+                    0
+                )
+
+                -- Whatever second widget would leave it if the third would be placed
+                size_third = math.max(size_remains - size_second, size_third)
+                place_widget(
+                    third,
+                    -- At the end
+                    size_total - size_third,
+                    -- Size calculated above - expand
+                    size_third
+                )
+
+                place_widget(
+                    second,
+                    -- After the first
+                    size_first,
+                    -- Whatever is left - shrink
+                    size_remains
+                )
             else
-                w, _ = base.fit_widget(self, context, self._private.second, size_second, height)
-                x = floor( (width -w)/2 )
+                place_widget(
+                    third,
+                    -- At the end
+                    size_total - size_third
+                    -- Takes the space it needs
+                )
+
+                -- Whatever second widget would leave it if the first would be placed
+                size_first = math.max(size_remains - size_second, size_first)
+                place_widget(
+                    first,
+                    -- At the beginning
+                    0,
+                    -- Size calculated above - expand
+                    size_first
+                )
+
+                place_widget(
+                    second,
+                    -- After the first
+                    size_first,
+                    -- Whatever is left - shrink
+                    size_remains
+                )
             end
+        else
+            -- Can place all widgets with the correct sizes
+
+            place_widget(
+                -- At the beginning
+                first,
+                0,
+                -- Same size as the largest widget - expand
+                max_size_side
+            )
+
+            place_widget(
+                third,
+                -- At the end
+                size_total - max_size_side,
+                -- Same size as the largest widget - expand
+                max_size_side
+            )
+
+            place_widget(
+                second,
+                -- After the first
+                max_size_side,
+                -- Whatever is left - expand
+                size_remains
+            )
         end
-        table.insert(result, base.place_widget_at(self._private.second, x, y, w, h))
+    else
```

Comment:
Can you add a comment here to explicitly say we default to INSIDE

Reply:
Sure, I will add the comment.